### ISSUE TITLE
Order polling too fast

### ISSUE THREAD

**Issue by robaman** [xenolf/lego#574]
Currently the polling for order status happens without any sort of delay. So if the acme client has a really fast connection to the ACME server it will hammer the server with as much HTTP requests as the network allows. If the ACME server has rate limits, the client will probably hit those.

What do you guys think about implementing a delay interval for this polling?

The section of the code I'm talking about is here: 
https://github.com/xenolf/lego/blob/d457f70ae0a39ea3e62ddad62caa7c29d33a01e2/acme/client.go#L711-L727

**Comment by mholt** [comment]
Hmm, good catch, we should probably have a time.Sleep here for... I dunno, maybe 1 second?

**Comment by robaman** [comment]
Yep. 1 second sounds about right to me.

**Comment by ldez** [comment]
I think a Ticker can be a solution.

**Comment by robaman** [comment]
Trying to understand Sleep vs Ticker. Can you explain the advantages @ldez ?

Here is a simple version with sleep in the mean time:
https://github.com/robaman/lego/commit/fbba98bd958e8da2fa22108fa26894ada4671fa7

**Comment by ldez** [comment]
I think a Ticker creates a more explicit code.
And this allows to define a time out which seems to me better than iterating 1000 times

In addition, one second of sleep can produce a wait of 16 minutes (1000 seconds) or more (related to HTTP calls), which seems too big.

```go
	// ...

	stopTimer := time.NewTimer(10 * time.Second)
	defer stopTimer.Stop()
	retryTick := time.NewTicker(200 * time.Millisecond)
	defer retryTick.Stop()

	for {
		select {
		case <-stopTimer.C:
			return nil, errors.New("timeout, operation failed")
		case <-retryTick.C:
			if _, err := getJSON(order.URL, &retOrder); err != nil {
				return nil, err
			}

			done, err := c.checkCertResponse(retOrder, &certRes, bundle)
			if err != nil {
				return nil, err
			}
			if done {
				return &certRes, nil
			}
		}
	}
}
```

**Comment by robaman** [comment]
I like it. Creating PR.

**Comment by robaman** [comment]
Created PR https://github.com/xenolf/lego/pull/581

**Comment by ldez** [comment]
Closed by  #581
